### PR TITLE
Provide unit tests for security manager

### PR DESCRIPTION
**Pull request by Kozoriz**
Added new mocks for correct working security_manager_tests
Fixed security_manager_tests, crypto_manager_tests
Added ssl_context_tests, ssl_sertificate_hand_shake_test

Related to APPLINK-20064

### CODE REVIEW

**Comment by Kozoriz** [comment]
please review : @malirod, @VVeremjova , @OHerasim, @pestOO

**Comment by AByzhynar** [comment]
Reviewed

**Comment by malirod** [comment]
Reviewed

File: src/components/security_manager/test/crypto_manager_impl_test.cc [hunk 1]
```diff
@@ -30,433 +30,193 @@
...
-    client_manager->Finish();
-    delete crypto_manager;
-    delete client_manager;
   }
-
-  virtual void SetUp() {
-    server_ctx = crypto_manager->CreateSSLContext();
-    client_ctx = client_manager->CreateSSLContext();
+  void TearDown() OVERRIDE { delete crypto_manager; }
+  void InitSecurityManger() {
```
Comment (VVeremjova) [inline src/components/security_manager/test/crypto_manager_impl_test.cc]: Correct name, please

**Comment by Kozoriz** [comment]
@VVeremjova  see f5cb565

**Comment by VVeremjova** [comment]
Reviewed

**Comment by OHerasym** [comment]
Reviewed

**Pull request merged**